logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -152,9 +151,10 @@ class BackgroundThreadTransport(Transport):
     def __init__(self, client, name):
         http = copy.deepcopy(client.connection.http)
         http = client.connection.credentials.authorize(http)
-        self.client = Client(client.project,
-                             client.connection.credentials,
-                             http)
+        # __class__ avoids a circular import
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2160
```

Comment:
You could also just not have circular dependencies. #2160 isn't really capturing what the problem is here. FWIW using `client.__class__` is just fine. So is use `type(client)`. (`__class__` is really only relevant for old-style classes, which cool kids never use, though @jonparrott does sometimes.)